Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx

```diff
@@ -0,0 +1,130 @@
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import { Button, Modal, ModalBody, ModalFooter } from "reactstrap";
+import { AppContext, IKeySerial } from "../../Types";
+import KeySerialAssignmentValues from "./KeySerialAssignmentValues";
+import KeySerialEditValues from "./KeySerialEditValues";
+import { toast } from "react-toastify";
+
+interface IProps {
+    isModalOpen: boolean;
+    closeModal: () => void;
+    openEditModal: (keySerial: IKeySerial) => void;
+    openUpdateModal: (keySerial: IKeySerial) => void;
+    onRevoke: (keySerial: IKeySerial) => void;
+    selectedKeySerial: IKeySerial;
+    updateSelectedKeySerial: (keySerial: IKeySerial) => void;
+
+}
+
+interface IState {
+    submitting: boolean;
+}
+
+export default class RevokeKeySerials extends React.Component<IProps, IState> {
+    public static contextTypes = {
+        fetch: PropTypes.func,
+        team: PropTypes.object
+    };
+
+    public context: AppContext;
+    constructor(props) {
+        super(props);
+        this.state = {
+            submitting: false
+        };
+    }
+
+    public componentDidMount() {
```

Comment:
we aren't fetching the data on the other revokes, but i'd wait to remove this until we get @srkirkland's opinion